### ISSUE TITLE
Use case findJobTimeRange.py can't find user-name

### ISSUE THREAD

**Issue by thanh-lam** [IBM/CAST#610]
**Describe the bug**
When I tried to run BDS use case script "findJobTimeRange.py", it seems to have a problem with the "user_name" format:
```
[root@bd01 python]# ./findJobTimeRange.py -a 526
# Found 1 matches for specified the job.

allocation-id: 526
job-id: 3119 - 0
Traceback (most recent call last):
  File "./findJobTimeRange.py", line 132, in <module>
    sys.exit(main(sys.argv))
  File "./findJobTimeRange.py", line 116, in main
    print( "user-name: {0} \nuser-id: {1}".format(tr_data["user_name"], tr_data["user_id"]))
KeyError: 'user_name'
```

Note that, allocation 526 was created most recently (March 13th). Allocations that were created in Feb. work fine with the command.

**To Reproduce**
1. Login to one of the BDS server
2. Change to dir /opt/ibm/csm/bigdata/python/
3. Run the command using a known allocation ID, for example 526:
     # ./findJobTimeRange.py -a 526

**Expected behavior**
The `findJobTimeRange.py` command should print out the begin-time and end-time of the allocation or job. Example:
```
[root@bd01 python]# ./findJobTimeRange.py -a 6870
# Found 1 matches for specified the job.

allocation-id: 6870
job-id: 22754 - 0
user-name: protocol 
user-id: 31415
begin-time: 2019-02-05.14:50:53:310 
end-time: now
```

**Screenshots**

**Environment (please complete the following information):**
 - Machine [IST BDS cluster: bd01, bd02]
 - Version [CSM 1.5.0-2137]

**Additional context**
It can be confirmed in kibana allocation indexes that:
- The allocation data created in March didn't have the `user-name` field.
- The allocation data created in Feb. did have the `user-name` field.

**Issue Source:**
A quick regression for BDS user cases.

**Comment by mew2057** [comment]
Okay so this one is easy/hard.  Basically, adding smt_mode re balanced the r/b tree backing the implementation of the transactional log. 

The r/b tree is statically computed and what happened is that the script where I serialized it had a bug where the children of non terminal nulls (a legal r/b case) weren't added to the serialization. Coincidentally this bug never showed up before, because the older balance was in fact fine.

I've got a candidate fix, but it will require a reinstall of csm-core.

**Comment by thanh-lam** [comment]
Verified the fix with CSM 1.5.1-2146. Now user name is in the kibana data:
```
...
      "job_submit_time": "2019-03-15 15:52:48",
      "requeue": "",
      "user_name": "protocol",
      "ssd_min": 0,
      "user_id": 31415,
...
```
And `findJobTimeRange.py` works:
```
# /opt/ibm/csm/bigdata/python/findJobTimeRange.py -a 583
# Found 1 matches for specified the job.

allocation-id: 583
job-id: 3278 - 0
user-name: protocol 
user-id: 31415
begin-time: 2019-03-15.15:52:50:435 
end-time: 2019-03-15.15:54:19:075
```
